exercise/templatetags/exercise.py

```diff
@@ -118,6 +118,81 @@ def submission_status(status):
     return Submission.STATUS[status]
 
 
+"""
+    Used for adding relevant classes to the
+    points_badge to enable textual status.
+"""
+@register.simple_tag(takes_context=True)
+def get_points_class(context):
```

Comment:
If you notice, the logic in both functions is the same, so one of them could provide it to the another one. E.g. `_get_points_type` that returns the class string or any other string. Then get_points_text can just use dict to map those strings to texts.

Reply:
Not sure if I managed to address everything you pointed out but I did make an attempt to consolidate the logic - it was not just as straightforward as I had thought. Anyway, you can have a look at the changes and see if there's (still) something to be fixed.